config/setup.go

```diff
@@ -150,6 +154,16 @@ func Configure(fname string, configSections string) (*SetupData, error) {
 			data.Config.Server.TLSKeyPath = ""
 		}
 
+		// If running in docker:
+		// 1. always bind to 0.0.0.0 instead of localhost
+		// 2. set paths of static files in UNIX manners
+		if !isDevEnv && isDocker {
+			data.Config.Server.TemplatesParentDir = "/usr/share/writefreely"
+			data.Config.Server.StaticParentDir = "/usr/share/writefreely"
+			data.Config.Server.PagesParentDir = "/usr/share/writefreely"
+			data.Config.Server.Bind = "0.0.0.0"
+		}
+
```

Comment:
Could we use a Docker-specific configuration file instead, or do you think this is the most straightforward way to way to handle this?

Reply:
I actually tried to do so by creating a template `config.ini` in `/data` directory inside the container with following content:

```
[server]
bind                 = 0.0.0.0
templates_parent_dir = /usr/share/writefreely
static_parent_dir    = /usr/share/writefreely
pages_parent_dir     = /usr/share/writefreely
```

which is equivalent to the lines added to `config/setup.go`.

But when running `writefreely config start`, writefreely seems read the (incomplete) `config.ini` file, and results in all the default values (e.g. DB filename `writefreely.db` if the user choice sqlite) omitted. **This would not provide an uniform user experience compared with non-docker setup**.

Thus, in my opinion, adding these lines may be the solution with the least modification to source code I could come up with.

Besides, these 4 lines are only effective when envvar `WRITEFREELY_DOCKER` is set, so probably incurs no impact on existing non-docker installs?

Reply:
Yep, that makes sense. Thanks for explaining!